backend/persistent/datastores/impala.js

```diff
@@ -15,6 +15,12 @@ export function createClient(connection) {
     return client;
 }
 
+export function connect(connection) {
+
+    // Runs a blank query to check connection has been established:
+    return createClient(connection).query('SELECT ID FROM (SELECT 1 ID) DUAL WHERE ID=0');
```

Comment:
Since it does not have a authenticate method, we can check if the connection succeeds by running a query which returns no results. This is the Impala equivalent of `dual` table -- i.e:
```
select 1 as A from dual where A=0
```